src/functional/iterator/transforms/pass_manager.py

```diff
@@ -40,15 +63,14 @@ def apply_common_transforms(
                 break
             ir = unrolled
             ir = NormalizeShifts().visit(ir)
-            if not use_tmps:
-                ir = InlineLifts().visit(ir)
+            ir = _inline_lifts(ir, lift_mode)
+            ir = NormalizeShifts().visit(ir)
         else:
             raise RuntimeError("Reduction unrolling failed.")
-    if use_tmps:
+    if lift_mode != LiftMode.FORCE_INLINE:
         assert offset_provider is not None
-        ir = CreateGlobalTmps().visit(
-            ir, offset_provider=offset_provider, register_tmp=register_tmp
-        )
+        ir = CreateGlobalTmps().visit(ir, offset_provider=offset_provider)
+        ir = InlineLifts().visit(ir)
 
     if common_subexpression_elimination:
         ir = CommonSubexpressionElimination().visit(ir)
```

Comment:
```suggestion
        ir = EtaReduction().visit(ir)
        ir = CommonSubexpressionElimination().visit(ir)
```
fixes the ICON stencil

Reply:
I put the EtaReduction outside of the if-condition as it is required by the gtfn backend. Could you double-check that all ICON stuff still works?

Reply:
Currently it compiled, but failed verification. I compared the generated code for one stencil and there were only name changes. Rerunning ICON test now.